common/cert-manager/cert-manager/base/kustomization.yaml

```diff
@@ -1,41 +1,16 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
-namespace: cert-manager
+
 resources:
-- namespace.yaml
-- api-service.yaml
-- cluster-role-binding.yaml
-- cluster-role.yaml
-- deployment.yaml
-- mutating-webhook-configuration.yaml
-- service-account.yaml
-- service.yaml
-- validating-webhook-configuration.yaml
-commonLabels:
-  kustomize.component: cert-manager
+- https://github.com/jetstack/cert-manager/releases/download/v1.3.0/cert-manager.yaml
```

Comment:
Should we use the download URL for cert-manager, or storing the actual files as before? It is easy to write kustomization if we can access these files in local IDE.

If we are going to use the external github link, should we move cert-manager to `contrib/` parent directory instead? Because it is third party tool.

Reply:
Personally I believe users shouldn't need to edit the actual cert-manager manifest, but if for some reason they need to customize it they should add an overlay or a PatchStrategicMerge, etc. I don't think the original YAML should be touched though, so linking to it upstream removes clutter and makes it easy to see what version is being used and to upgrade it.

This is also the setup I am using for all the Kubeflow components in the Argo CD deployment, which makes it a lot easier for users as they don't need to deal with the huge amount of YAML. https://github.com/kubeflow-onprem/ArgoFlow

Reply:
Thank you David for the explanation! I understand the benefit. I agree that there are pros and cons choosing one approach versus the other. So I will leave it to you to decide.

Besides, I cannot access this link https://github.com/kubeflow-onprem/ArgoFlow, is this the right link or is it private repo?

cc @Bobgy

Reply:
@zijianjoy Yeah the repo needed to change homes suddenly. It's now https://github.com/argoflow/argoflow.